src/tool_operate.h

```diff
@@ -41,6 +41,7 @@ struct per_transfer {
   struct metalinkfile *mlfile;
   struct metalink_resource *mlres;
   char *this_url;
+  int urlnum; /* the index of the given URL */
```

Comment:
Does this need to be an int and signed?

Reply:
Good remark. I'll at least make it unsigned. I would think that a 32 bit counter is large enough?